examples/features/xds/server/main.go

```diff
@@ -72,65 +66,40 @@ func determineHostname() string {
 	return hostname
 }
 
-func init() {
-	flag.Usage = func() {
-		fmt.Fprintf(flag.CommandLine.Output(), `
-Usage: server [port [hostname]]
-
-  port
-        The listen port. Defaults to %d
-  hostname
-        The name clients will see in greet responses. Defaults to the machine's hostname
-`, defaultPort)
-
-		flag.PrintDefaults()
-	}
-}
-
 func main() {
 	flag.Parse()
-	if *help {
-		flag.Usage()
-		return
-	}
-	args := flag.Args()
 
-	if len(args) > 2 {
-		flag.Usage()
-		return
+	greeterPort := fmt.Sprintf(":%d", *port)
+	greeterLis, err := net.Listen("tcp4", greeterPort)
+	if err != nil {
+		log.Fatalf("net.Listen(tcp4, %q) failed: %v", greeterPort, err)
 	}
 
-	port := defaultPort
-	if len(args) > 0 {
+	creds := insecure.NewCredentials()
+	if *xdsCreds {
+		log.Println("Using xDS credentials...")
 		var err error
-		port, err = strconv.Atoi(args[0])
-		if err != nil {
-			log.Printf("Invalid port number: %v", err)
-			flag.Usage()
-			return
+		if creds, err = xdscreds.NewServerCredentials(xdscreds.ServerOptions{FallbackCreds: insecure.NewCredentials()}); err != nil {
+			log.Fatalf("failed to create server-side xDS credentials: %v", err)
 		}
 	}
 
-	var hostname string
-	if len(args) > 1 {
-		hostname = args[1]
-	}
-	if hostname == "" {
-		hostname = determineHostname()
-	}
+	greeterServer := xds.NewGRPCServer(grpc.Creds(creds))
+	pb.RegisterGreeterServer(greeterServer, &server{serverName: determineHostname()})
 
-	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	healthPort := fmt.Sprintf(":%d", *port+1)
+	healthLis, err := net.Listen("tcp4", healthPort)
 	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
+		log.Fatalf("net.Listen(tcp4, %q) failed: %v", healthPort, err)
 	}
-	s := grpc.NewServer()
-	pb.RegisterGreeterServer(s, newServer(hostname))
-
-	reflection.Register(s)
+	grpcServer := grpc.NewServer()
 	healthServer := health.NewServer()
 	healthServer.SetServingStatus("", healthpb.HealthCheckResponse_SERVING)
-	healthpb.RegisterHealthServer(s, healthServer)
+	healthpb.RegisterHealthServer(grpcServer, healthServer)
```

Comment:
Why is health on a different server? Because of creds?

Reply:
Yes, because the `greeterServer` is configured to use xDS credentials on its serving port, we cannot export the health service on the same port since the GCP health checker will not have the required credentials.

This pattern of using `serving_port + 1` for the health service is mentioned in the UG and I'm guessing this is what the PSM security interop tests would do as well.

The other option is to use TCP health checks if we want to use a single port. But using gRPC health checks on `serving_port + 1` was deemed a better option that the former.